tests/main/auto-refresh-gating-from-snap/task.yaml

```diff
@@ -0,0 +1,129 @@
+summary: Check that snapctl refresh --pending/--proceed can be used outside of
+  hooks.
+
+details: |
+  Test auto-refresh with gate-auto-refresh hook support enabled
+  (experimental.gate-auto-refresh-hook feature) and verify the snaps can
+  use snapctl refresh --pending/--proceed commands outside of hooks.
+  The test uses two test snaps, there are a few versions of these
+  snaps in the store (in stable/beta/edge channels) for this test.
+
+environment:
+  SNAP_NAME: test-snapd-refresh-control
+  CONTENT_SNAP_NAME: test-snapd-refresh-control-provider
+  CONTROL_FILE: /var/snap/test-snapd-refresh-control/common/control
+  DEBUG_LOG_FILE: /var/snap/test-snapd-refresh-control/common/debug.log
+
+prepare: |
+  snap install --devmode jq
+  snap set system experimental.gate-auto-refresh-hook=true
+
+debug: |
+  jq -r '.data["snaps-hold"]' < /var/lib/snapd/state.json || true
+
+execute: |
+  force_autorefresh() {
```

Comment:
I think this should be paft of a tool, I'll create another or with this. I see it is used in several tests.

Reply:
Indeed, excellent.